src/modules/ekf2/EKF/gps_control.cpp

```diff
@@ -379,10 +370,20 @@ void Ekf::controlGpsYawFusion(const gpsSample &gps_sample, bool gps_checks_passi
 		} else {
 			if (starting_conditions_passing) {
 				// Try to activate GPS yaw fusion
-				startGpsYawFusion(gps_sample);
+				if (resetYawToGps(gps_sample.yaw)) {
```

Comment:
Why moving all the implementations details here? I find it more readable if we just call `startGpsYawFusion`, even if we don't use that function at several places.

Reply:
Because it can fail, so rather than blindly call "start" then after check the control status flag, we only start it if the reset succeeds. Explicitly stopping EV yaw and mag can also go away later, we can have them stop on their own.

If we can get these things broken out of the sprawling Ekf superclass I'd be a lot more comfortable with little private helper methods refactored for readability.